cns/configuration/configuration.go

```diff
@@ -35,6 +35,7 @@ type CNSConfig struct {
 	EnableSubnetScarcity        bool
 	EnableSwiftV2               bool
 	InitializeFromCNI           bool
+	EnableHomeAz                bool
```

Comment:
Maybe this should be `EnableNNCCreation`. It will be clearer when AKS reviews toggles to CNS config.

Reply:
We typically control the component configurations closer to our service feature naming so keeping it related to HomeAz makes more sense as the flag is to use the HomeAzMonitor overall to retrieve the Az and then create the NNC from that.

Reply:
I know that "Az" is everywhere, so it's probably impractical to change at this point, but "AZ" is an abbreviation, so this should be `EnableHomeAZ` by typical Go naming conventions.

Reply:
Changed this to EnableHomeAZ